flink-table/flink-sql-parser/src/test/java/org/apache/flink/sql/parser/FlinkDDLValidationTest.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.sql.parser;
+
+import org.apache.flink.sql.parser.ddl.SqlCreateTable;
+import org.apache.flink.sql.parser.ddl.SqlCreateView;
+import org.apache.flink.sql.parser.error.SqlValidateException;
+import org.apache.flink.sql.parser.impl.FlinkSqlParserImpl;
+
+import org.apache.calcite.avatica.util.Casing;
+import org.apache.calcite.avatica.util.Quoting;
+import org.apache.calcite.sql.SqlNodeList;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.SqlParserImplFactory;
+import org.apache.calcite.util.SourceStringReader;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Validation tests for ddl fields.
+ */
+public class FlinkDDLValidationTest {
+
+	protected SqlParserImplFactory parserImplFactory() {
+		return FlinkSqlParserImpl.FACTORY;
+	}
+
+	private SqlParser getSqlParser(String sql) {
+		SourceStringReader source = new SourceStringReader(sql);
+		return SqlParser.create(source,
+			SqlParser.configBuilder()
```

Comment:
In `FlinkSqlParserImplTest` there is a `ValidationMatcher` which can be used to validate failed cases. The case you gave are all positive tests, i think with code `sql(sqlStr).node(new ValidationMatcher());` you can trigger the validation, so you can move these test cases into `FlinkSqlParserImplTest`.

Reply:
I've deleted the FlinkDDLValidationTest.java and migrate the tests into FlinkSqlParserImplTest.

BTW, I find there is a serious bug in ValidationMatcher in FlinkSqlParserImplTest. I've created a jira issue(https://issues.apache.org/jira/browse/FLINK-14384) for it.

Reply:
Yes, it is, we may add an exception assertion when we passed in the error message explicitly.

Reply:
+1. We can trace this problem in the JIRA issue. Resolving this...